src/eventListeners.js

```diff
@@ -0,0 +1,51 @@
+import { Rectangle } from './shapes/Rectangle';
+import { Polygon } from './shapes/Polygon';
+
+/* All event listener callbacks go in this file */
+
+/* --- RECTANGLE EVENT LISTENERS --- */
+export function onMouseClickRectangle() {
+    const rect = this.canvas.currentShape;
+    if (this.canvas.clicked) {
+        this.canvas.add(new Rectangle(this.canvas));
+        this.editEventListenerInvokeStatus("mousemove", true);
+    } else {
+        this.editEventListenerInvokeStatus("mousemove", false);
+        rect.registerClick(this.canvas.getCursorPosition(event));
+    }
+}
+
+export function onMouseMoveRectangle(event) {
+    const rect = this.canvas.currentShape;
+    rect.registerClick(this.canvas.getCursorPosition(event));
+}
+/* --- END --- */
+
+/* --- POLYGON EVENT LISTENERS ---*/
+export function onMouseClickPolygon(event) {
+    if (this.canvas.clicked) {
+        this.canvas.add(new Polygon(this.canvas));
+        this.editEventListenerInvokeStatus("mousemove", true);
+    } else {
+        const polygon = this.canvas.currentShape;
+        const currentClickPosition = this.canvas.getCursorPosition(event);
+        const lastClickPosition = polygon.getLastClickPosition();
+
+        if (currentClickPosition.equals(polygon.initialCursorPosition) || currentClickPosition.equals(lastClickPosition)) {
```

Comment:
should it be a falloff radius instead of `equals`

Reply:
yup it should be a falloff radius.

next pr